qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -87,12 +81,16 @@ def run(self, dag):
         num_cx = 0
         num_meas = 0
 
-        # Get avg number of cx and meas per qubit
-        ops = dag.count_ops()
-        if "cx" in ops.keys():
-            num_cx = ops["cx"]
-        if "measure" in ops.keys():
-            num_meas = ops["measure"]
+        if self.target is not None:
+            num_cx = 1
+            num_meas = 1
+        else:
+            # Get avg number of cx and meas per qubit
+            ops = dag.count_ops()
+            if "cx" in ops.keys():
+                num_cx = ops["cx"]
+            if "measure" in ops.keys():
+                num_meas = ops["measure"]
```

Comment:
It seems to change the logic, is it intended? The original code counts the number of ops in "circuit" while the new code sets the constants (~= the number of ops defined in "target"?).

Reply:
Yeah it was an intentional chamge, see the commit message on: https://github.com/Qiskit/qiskit-terra/pull/7776/commits/4fe2c44c8915a623a1ac668ffabd892c8ebdb402 It's mainly because at the layout stage there isn't an easy way to determine the ratio of 1q to 2q operations in the circuit besides iterating over the entire circuit and counting (which adds a lot of overhead). Also so early in the pass manager I'm not sure how accurate scaling the error based on the number of gates will be because we haven't done routing, basis translation, or any optimization passes. Since it's just a heuristic I figured just using the raw error rates was sufficient.

But I definitely am not 100% sure on this, if you have a suggestion for a better way to do this  I'm definitely open to making this better. The 2q error rates weren't even working before #7740 (which fixed the bug) so we can take this opportunity to come up with a better heuristic for weighing the error of the layouts.